src/BaseVirtualComponent.jsx

```diff
@@ -39,129 +69,170 @@ function propsDiffer(propsA, propsB) {
-                    items.push(item);
-                }
-                else {
-                    items = items.concat(item.children);
-                }
-            }
-        }
-        return items || this._items;
+    /**
+     * Overrides the renderChildren() of BaseComponent, so it can inject the children from the
+     * linked component, if it's linked!
+     */
+    renderChildren(name, args) {
+        return this[_linked].component
+            ? this[_linked].component.renderChildren(name, args)
+            : super.renderChildren(name, args);
     }
 
-    forEach(fn) {
-        this.children.forEach(fn);
+    /**
+     * By default we just render whatever children were passed in - this can be overridden,
+     * e.g. if conditional logic is needed or you want to add additional virtual components
+     */
+    render() {
+        return this.renderChildren() || null;
     }
 
-    any(fn) {
-        return this.children.some(fn);
+    /**
+     * This is the initialization code for a virtual component. Typically a virtual component exists
+     * alongside a concrete component, as a mechanism for configuring that component. For example,
+     * in a virtual scroller, the concrete component will render actual elements to the DOM (based on
+     * what's visible), but its children will be rendered to a virtual tree for doing the layout.
+     *
+     * A virtual component can't be rendered by ReactDOM, so instead, you need to pass the children
+     * of the concrete component to a virtual component, that will render it instead. You only need
+     * to call bindToComponent on the root virtual component - this just tells the virtual component
+     * to monitor the children of the given component, and render itself whenever they change.
+     *
+     * This method returns the virtual component, so that you can write (from a component):
+     *
+     * ```
+     * this.virtualComponent = new MyVirtualComponent().linkToComponent(this);
+     * ```
+     *
+     * @param {BaseComponent} component The component to bind to.
+     * @returns {BaseVirtualComponent} This virtual component.
+     */
+    linkToComponent(component) {
+        if (!(component instanceof BaseComponent)) {
+            throw new Error('@VirtualComponent.linkToComponent() expects an @Component as its argument');
+        }
+
+        this[_linked].component = component;
+        this.forceUpdate();
+        return component.link(this);
     }
 
-    map(fn) {
-        return this.children.map(fn);
+    /**
+     * The virtual children of the virtual component.
+     * @type {Array.<BaseVirtualComponent>}
+     */
+    get children() {
+        return flatten(this[_items], isNullOrNotVirtual);
     }
 
-    filter(fn) {
-        return this.children.filter(fn);
+    /**
+     * Shorthand for iterating over the children of the virtual component.
+     * @param {function} callback Function to call on each child.
+     */
+    forEach(fn) {
```

Comment:
Do we need to have these helper methods (forEach and hasChildren)? I'd be inclined to remove them unless they're providing a useful abstraction.

Reply:
I removed most of them, except for forEach, which is commonly used (e.g. for iterating layout). But you're right, it's not really necessary, since we'll be rewriting the virtual scroller anyway.